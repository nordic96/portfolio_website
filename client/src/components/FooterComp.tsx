import React from 'react';
import EmailIcon from '../assets/images/email-icon.png';

/**
 * Footer Component for Main Page (contains footer information)
 */
function FooterComp() {
    return (
        <div className={'text-white px-24 py-8 bg-slate-800'}>
            <div className={'flex flex-row justify-between gap-4'}>
                <div className={'flex-1 text-left'}>
                    <p className={'font-bold text-lg'}>Contacts</p>
                    <div className={'flex flex-row gap-2'}>
                        <img
                            className="contact-icon"
                            src={EmailIcon}
                            alt="email"
                        />
                        <p className={'text-base'}>
                            rhrlgns96@gmail.com (personal)
                        </p>
                    </div>
                </div>
                <div className={'flex-1 text-left'}>
                    <p className={'font-bold text-lg'}>About this Website</p>
                    <p className={'text-base'}>
                        This project was built using NodeJS, ReactJS, deployed
                        in Heroku App Server. Project information was pulled
                        from MongoDB Server
                    </p>
                </div>
            </div>
        </div>
    );
}
export default React.memo(FooterComp);
