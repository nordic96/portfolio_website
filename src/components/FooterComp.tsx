import LabelContainer from 'labelcontainer';
import React from 'react';

/**
 * Footer Component for Main Page (contains footer information)
 */
function FooterComp() {
    const lsInstance = LabelContainer.getInstance();
    return (
        <div
            className={
                'relative text-white py-4 bg-coolblack bottom-0 w-full h-32 flex justify-center'
            }>
            <div className={'flex flex-row justify-between gap-4 w-[72rem]'}>
                <div className={'flex-1 text-left'}>
                    <p
                        className={
                            'font-bold lg:text-lg md:text-base max-sm:text-sm'
                        }>
                        About this Website
                    </p>
                    <p className={'lg:text-base md:text-sm max-sm:text-xs'}>
                        {lsInstance.getLabel('footer_desc_1')}
                    </p>
                </div>
            </div>
        </div>
    );
}
export default React.memo(FooterComp);
