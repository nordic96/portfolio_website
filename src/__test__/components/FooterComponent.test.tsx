import React from 'react';
import { render } from '@testing-library/react';
import FooterComp from '../../components/FooterComp';

describe('Footer Component', () => {
    it('should match with snapshot', () => {
        const footer = render(<FooterComp />);
        expect(footer).toMatchSnapshot();
    });
});
