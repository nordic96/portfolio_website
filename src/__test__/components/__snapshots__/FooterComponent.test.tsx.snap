// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative text-white py-4 bg-coolblack bottom-0 w-full h-32 flex justify-center"
      >
        <div
          class="flex flex-row justify-between gap-4 lg:w-[70.5rem] max-sm:px-4"
        >
          <div
            class="flex-1 text-left"
          >
            <p
              class="font-bold lg:text-lg md:text-base max-sm:text-sm"
            >
              About this Website
            </p>
            <p
              class="lg:text-base md:text-sm max-sm:text-xs"
            >
              ©2025 Stephen Ko All Rights Reserved. Website design and development adhere to accessibility guidelines.
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative text-white py-4 bg-coolblack bottom-0 w-full h-32 flex justify-center"
    >
      <div
        class="flex flex-row justify-between gap-4 lg:w-[70.5rem] max-sm:px-4"
      >
        <div
          class="flex-1 text-left"
        >
          <p
            class="font-bold lg:text-lg md:text-base max-sm:text-sm"
          >
            About this Website
          </p>
          <p
            class="lg:text-base md:text-sm max-sm:text-xs"
          >
            ©2025 Stephen Ko All Rights Reserved. Website design and development adhere to accessibility guidelines.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
